app/api/utils.py

```diff
@@ -242,45 +243,51 @@ class CoNLLParser(FileParser):
     ```
     """
     def parse(self, file):
-        words, tags = [], []
         data = []
         file = io.TextIOWrapper(file, encoding='utf-8')
-        for i, line in enumerate(file, start=1):
-            if len(data) >= settings.IMPORT_BATCH_SIZE:
-                yield data
-                data = []
-            line = line.strip()
-            if line:
-                try:
-                    word, tag = line.split('\t')
-                except ValueError:
-                    raise FileParseException(line_num=i, line=line)
+
+        # Add check exception
+
+        field_parsers = {
+            "ne": lambda line, i: conllu.parser.parse_nullable_value(line[i]),
+        }
+
+        try:
+            sentences = conllu.parse(
+                file.read(),
+                fields=("form", "ne"),
+                field_parsers=field_parsers
+            )
+        except conllu.parser.ParseException as e:
+            raise FileParseException(line_num=-1, line=str(e))
+
+        for sentence in sentences:
+            if not sentence:
+                continue
+            # if len(data) >= settings.IMPORT_BATCH_SIZE:
+            #     yield data
+            #     data = []
+            words, labels = [], []
+            for item in sentence:
+                word = item.get("form")
+                tag = item.get("ne", None)
```

Comment:
Nit: `.get('ne', None)` should be equivalent to `.get('ne')` for any normal dictionary.